Step2

hyoj opened the pull request:
안녕하세요.
Step2 리뷰 부탁드립니다!

hyoj committed:
Step2 : 질문 CRUD 기능 ATDD 기반으로 구현

hyoj committed:
Step2 : Question 단위 테스트 추가

jihan805 commented on src/test/java/nextstep/domain/QuestionTest.java:
```diff
@@ -0,0 +1,38 @@
+import support.test.BaseTest;
+
+public class QuestionTest extends BaseTest {
+    public static final User WEATHERMAN = new User(1L, "weatherman", "password", "wea man", "today@weather.com");
```
Question에 대한 테스트와  QuestionAcceptanceTest 작성 👍 
1단계에서 UserService에 대한 테스트를 구현하였듯 QuestionService에 대한 테스트도 구현해보는 것은 어떨까요!?

jihan805 commented on src/main/java/nextstep/service/QnaService.java:
```diff
@@ -28,22 +32,33 @@
+        original.update(loginUser, updatedQuestion);
+
+        return questionRepository.save(original);
     }
```
메소드에 @ Transactional 애노테이션이 있는데, questionRepository.save(..)와 같이 명시적으로 구현하지 않고 단순히 question.update(questionDto) 만으로도 수정이 가능한지 테스트해 보면 좋을것 같아요!

jihan805 commented on src/main/java/nextstep/service/QnaService.java:
```diff
@@ -28,22 +32,33 @@
+        }
+
+        questionRepository.delete(deletedQuestion);
     }
```
update 구현과 같이 삭제 상태를 변경하기 위해 자신의 글인지 체크하는 로직을 Question에 구현하는 것은 어떨까요?
더불어 현재는 물리적 삭제 처리를 하고 있는데 Question의 private boolean deleted 라는 필드를 활용해보는 것도 좋을것 같아요!

jihan805 reviewed (changes requested):
안녕하세요! qna 2단계 구현에 고생 많으셨어요!
전체적으로 정말 잘 구현하셨어요 👍 
몇가지 피드백을 남겼으니 확인 부탁드릴게요.
감사합니다 :)

hyoj committed:
Step2 로그아웃 ATDD 기반 구현

hyoj committed:
Step2 로그인 실패 페이지 ATDD 구현

- 로그인 폼, 로그인 실패 폼 URL 로 로그인 한 상태로 접근하면 메인 페이지로 리다이렉트
- HttpSessionUtils 세션에 저장된 로그인한 사용자가 없을 경우 GUEST_USER 반환하도록 수정
- 로그인 실패 html 파일명 카멜 케이스로 변경

hyoj committed:
Step2 폼 데이터 빌더에서 사용되는 HTTP METHOD 상수화

hyoj committed:
코드 정렬

hyoj committed:
Step2 테스트 데이터 삽입 구문 수정

hyoj committed:
Step2 질문 수정/삭제 기능 수정

- 질문 수정/삭제 기능 테스트를 위한 QnaServiceTest 추가

- Question 객체 자신이 삭제 가능한 상태인지 판단하도록 수정

- 실제 데이터 갱신은 각 entity 에서 처리하므로,
  QnaService 에서 수정/삭제 수행시 Repository 메서드 호출하지 않도록 수정

jihan805 reviewed (approved):
안녕하세요! 
2단계 피드백 반영 감사합니다 👍 잘하셨어요!!
머지 할게요, 다음 단계 진행 부탁드려요 :)
